move/committee/sources/committee.move

```diff
@@ -0,0 +1,351 @@
+
+module seal_committee::seal_committee;
+
+use seal_testnet::key_server::{
+    KeyServer,
+    create_partial_key_server,
+    create_committee_v2,
+    PartialKeyServer
+};
+use std::{string::String, u64};
+use sui::{transfer::Receiving, vec_map::{Self, VecMap}, vec_set::{Self, VecSet}};
+
+// ===== Errors =====
+
+const ENotMember: u64 = 0;
+const EInvalidMembers: u64 = 1;
+const EInvalidThreshold: u64 = 2;
+const EInsufficientOldMembers: u64 = 3;
+const EAlreadyRegistered: u64 = 4;
+const ENotRegistered: u64 = 5;
+const EAlreadyProposed: u64 = 6;
+const EInvalidProposal: u64 = 7;
+const EInvalidState: u64 = 8;
+
+// ===== Structs =====
+
+/// Member information to register with two public keys and the key server URL.
+public struct MemberInfo has copy, drop, store {
+    /// ECIES encryption public key, used during offchain DKG.
+    enc_pk: vector<u8>,
+    /// Signing PK, used during offchain DKG.
+    signing_pk: vector<u8>,
+    /// URL that the partial key server is running at.
+    url: String,
+}
+
+/// Valid states of the committee that holds state specific infos.
+public enum State has drop, store {
+    Init {
+        members_info: VecMap<address, MemberInfo>,
+    },
+    PostDKG {
+        members_info: VecMap<address, MemberInfo>,
+        partial_pks: vector<vector<u8>>,
+        pk: vector<u8>,
+        approvals: VecSet<address>,
+    },
+    Finalized,
+}
+
+/// MPC committee with defined threshold and members with its state.
+public struct Committee has key {
+    id: UID,
+    threshold: u64,
+    /// The members of the committee. The 'party_id' used in the DKG protocol is the index of this
+    /// vector.
+    members: vector<address>,
+    state: State,
+    /// Old committee ID that this committee rotates from.
+    old_committee_id: Option<ID>,
+}
+
+// ===== Public Functions =====
+
+/// Create a committee for fresh DKG with a list of members and threshold. The committee is in Init
+/// state with empty members_info.
+public fun init_committee(threshold: u64, members: vector<address>, ctx: &mut TxContext) {
+    init_internal(threshold, members, option::none(), ctx)
+}
+
+/// Create a committee for rotation from an existing finalized old committee. The new committee must
+/// contain an old threshold of the old committee members.
+public fun init_rotation(
+    old_committee: &Committee,
+    threshold: u64,
+    members: vector<address>,
+    ctx: &mut TxContext,
+) {
+    // Verify the old committee is finalized for rotation.
+    assert!(old_committee.is_finalized(), EInvalidState);
+
+    // Check that new committee has at least the threshold of old committee members.
+    let mut continuing_members = 0;
+    members.do!(|member| if (old_committee.members.contains(&member)) {
+        continuing_members = continuing_members + 1;
+    });
+    assert!(continuing_members >= (old_committee.threshold), EInsufficientOldMembers);
+
+    init_internal(threshold, members, option::some(old_committee.id()), ctx);
+}
```

Comment:
cc @benr-ml do we want to enforce the admin to be part of the committee members proposed? 

similar for rotation, it should be part of the new committee?